core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/AssetTransferFlows.kt

```diff
@@ -0,0 +1,859 @@
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val role = session.receive<AssetTransferResponderRole>().unwrap { it }
+            if (role == AssetTransferResponderRole.ISSUER) {
+                val signTransactionFlow = object : SignTransactionFlow(session) {
+                    override fun checkTransaction(stx: SignedTransaction) = requireThat {
+                    }
+                }
+                try {
+                    val txId = subFlow(signTransactionFlow).id
+                    println("Issuer signed transaction.")
+                    return subFlow(ReceiveFinalityFlow(session, expectedTxId = txId))
+                } catch (e: Exception) {
+                    println("Error signing unlock asset transaction by Issuer: ${e.message}\n")
+                    return subFlow(ReceiveFinalityFlow(session))
+                }
+            } else if (role == AssetTransferResponderRole.OBSERVER) {
+                val sTx = subFlow(ReceiveFinalityFlow(session, statesToRecord = StatesToRecord.ALL_VISIBLE))
+                println("Received Tx: ${sTx}")
+                return sTx
+            } else {
+                println("Incorrect Responder Role.")
+                throw IllegalStateException("Incorrect Responder Role.")
+            }
+        }
+    }
+
+}
+
+/**
+ * The IsAssetPledged flow is used to check if an asset is pledged or not.
+ *
+ * @property pledgeId The unique identifier representing the pledge on an asset for transfer, in the exporting n/w.
+ */
+@StartableByRPC
+class IsAssetPledged(
+    val pledgeId: String
+) : FlowLogic<Boolean>() {
+    /**
+     * The call() method captures the logic to fetch [AssetPledgeState] associated with input pledgeId.
+     *
+     * @return Returns boolean value, true or false.
+     */
+    @Suspendable
+    override fun call(): Boolean {
+        val linearId = getLinearIdFromString(pledgeId)
+        println("Getting [AssetPledgeState] for linearId: $linearId.")
+        val states = serviceHub.vaultService.queryBy<AssetPledgeState>(
+            QueryCriteria.LinearStateQueryCriteria(linearId = listOf(linearId))
+        ).states
+        if (states.isEmpty()) {
+            println("No [AssetPledgeState] state found associated with pledgeId: $pledgeId.")
+            return false
+        } else {
+            val pledgedState = states.first().state.data
+            println("Obtained [AssetPledgeState] state: $pledgedState.")
+            if (Instant.now().isAfter(Instant.ofEpochSecond(pledgedState.expiryTimeSecs))) {
+                return false
+            }
+            return true
+        }
+    }
+}
+
+/**
+ * The GetAssetPledgeStateById flow is used to fetch the asset pledge details (from export n/w).
+ *
+ * @property pledgeId The unique identifier representing the pledge on an asset for transfer, in the exporting n/w.
+ */
+@StartableByRPC
+class GetAssetPledgeStateById(
```

Comment:
This is redundant function that we decided to remove right?

Reply:
I think I have already removed that. This one is not redundant Sandeep.

Reply:
So what utility does this provide, that `GetAssetPledgeStatus` flow doesn't?

Reply:
`GetAssetPledgeStateById` returns `StateAndRef` and is used inside the `ReclaimPledgedAsset` flow, while `GetAssetPledgeStatus` returns `State` and is used in the intro-query.

Reply:
I thought we decided to convert this to a utility function and call it in `ReclaimPledgedAsset` and `GetAssetPledgeStatus` to get the pledge state.

Reply:
As I have commented above, these 2 functions `GetAssetPledgeStateById` and `GetAssetPledgeStatus` are returning different values (which I suppose we have overlooked earlier). The only common part in these these two functions is the below statement
```
val states = serviceHub.vaultService.queryBy<AssetPledgeState>(
            QueryCriteria.LinearStateQueryCriteria(linearId = listOf(linearId))
        ).states
```
Creating a new function for the above statement, and replacing the above statement in those 2 functions with this new function name is not required, I feel. Hence didn't make any change here Sandeep.

Reply:
Ok This can remain here as utility function, no issues, but I'd suggest not to use subFlow call (same reason being expensive) to this flow, to get StateAndRef in other flows, like Reclaim Flow (which we identified in last call, and decided to change to function call for performance reasons).